### ISSUE TITLE
[Translatable] Detect untranslated entry for translated property/entity

### ISSUE THREAD

**Issue by mmoll** [doctrine-extensions/DoctrineExtensions#2266]
Let's say I have the following entity:
```php
<?php

declare(strict_types=1);

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Gedmo\Mapping\Annotation as Gedmo;
use Gedmo\Translatable\Translatable;

/**
 * @ORM\Entity
 *
 * @Gedmo\TranslationEntity(class="App\Entity\Translation\GreetingTranslation")
 */
class Greeting implements Translatable
{
[...]
    /**
     * @ORM\Column
     *
     * @Gedmo\Translatable
     */
    #[Assert\NotBlank]
    public string $name = '';
}
```

and the following TranslationEntity:

```php
<?php

declare(strict_types=1);

namespace App\Entity\Translation;

use Doctrine\ORM\Mapping as ORM;
use Gedmo\Translatable\Entity\MappedSuperclass\AbstractTranslation;

/**
 * @ORM\Table(name="greeting_translations", indexes={
 *      @ORM\Index(name="greeting_translations_idx", columns={"locale", "object_class", "field", "foreign_key"})
 * })
 * @ORM\Entity(repositoryClass="Gedmo\Translatable\Entity\Repository\TranslationRepository")
 */
class GreetingTranslation extends AbstractTranslation
{
}
```
This is working all fine with `fallback = false`. I can have entities in multiple locales and also have entries per entity that are not existant in all locales. When retrieving the entity for a locale that doesn't have a translation, the following can be seen: `"Cannot assign null to property App\\Entity\\Greeting::$name of type string"`. I tried to add a custom exception [in PostLoad()](https://github.com/doctrine-extensions/DoctrineExtensions/blob/8c02cee09e3dd43799ec3b84b619b19982c47f43/src/Translatable/TranslatableListener.php#L471), however, this breaks all tests.

Basically, because of `|| (!$this->translationFallback && (!isset($config['fallback'][$field]) || !$config['fallback'][$field]))` the code after the condition always gets excuted and I'm a bit puzzled, why there's no exception for this case already there.

How would it be possible with the current code to add a new exception to this extension in order to catch it in the application for the case if there's an Entity, but for the selected locale that entry is not translated?

**Comment by franmomu** [comment]
Maybe https://github.com/doctrine-extensions/DoctrineExtensions/pull/2267 would fix this.

**Comment by mmoll** [comment]
With the current way, the extension works, this is not possible.
